File: Python/950. Reveal Cards In Increasing Order.py
from collections import deque


class Solution:
    def deckRevealedIncreasing(self, deck):
        n = len(deck)
        ans = [0] * n
        deck.sort()
        indices = deque(range(n))

        for card in deck:
            ans[indices.popleft()] = card
            if indices:
                indices.append(indices.popleft())

        return ans

File: C++/950. Reveal Cards In Increasing Order.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck)
    {
        vector<int> ans(deck.size(), 0);
        sort(deck.begin(), deck.end());
        queue<int> q;

        for (int i = 0; i < deck.size(); i++)
            q.push(i);

        for (int i = 0; i < deck.size(); i++)
        {
            ans[q.front()] = deck[i];
            q.pop();
            q.push(q.front());
            q.pop();
        }

        return ans;
    }
};